packages/api/src/jobs/update_just_read_feed.ts

```diff
@@ -0,0 +1,397 @@
+  author: item.author,
+  dir: item.dir || 'ltr',
+  date: item.createdAt,
+  topic: item.topic,
+  wordCount: item.wordCount!,
+  siteIcon: item.siteIcon,
+  publishedAt: item.publishedAt,
+  subscription: {
+    name: item.source.name,
+    type: item.source.type,
+  },
+})
+
+const selectCandidates = async (user: User): Promise<Array<Candidate>> => {
+  const userId = user.id
+  // get last 100 library items saved and not seen by user
+  const libraryItems = await searchLibraryItems(
+    {
+      size: 100,
+      includeContent: false,
+      query: `-is:seen wordsCount:>0`,
+    },
+    userId
+  )
+
+  logger.info(`Found ${libraryItems.length} library items`)
+
+  // get subscriptions for the library items
+  const subscriptionNames = libraryItems
+    .filter((item) => !!item.subscription)
+    .map((item) => item.subscription as string)
+
+  const subscriptions = await findSubscriptionsByNames(
+    userId,
+    subscriptionNames
+  )
+
+  // map library items to candidates and limit to 70
+  const privateCandidates: Array<Candidate> = libraryItems
+    .map((item) => libraryItemToCandidate(item, subscriptions))
+    .slice(0, 70)
+  const privateCandidatesSize = privateCandidates.length
+
+  logger.info(`Found ${privateCandidatesSize} private candidates`)
+
+  // get 100 items not seen by the user from public inventory
+  const publicItems = await findUnseenPublicItems(userId, {
+    limit: 100,
+  })
+
+  logger.info(`Found ${publicItems.length} public items`)
+
+  // map public items to candidates and limit to the remaining vacancies
+  const publicCandidates: Array<Candidate> = publicItems
+    .map(publicItemToCandidate)
+    .slice(0, 100 - privateCandidatesSize)
+
+  const publicCandidatesSize = publicCandidates.length
+  logger.info(`Found ${publicCandidatesSize} public candidates`)
+
+  // return 100 candidates and 70 from private and 30 from public
+  return [...privateCandidates, ...publicCandidates]
+}
+
+const rankCandidates = async (
+  userId: string,
+  candidates: Array<Candidate>
+): Promise<Array<Candidate>> => {
+  if (candidates.length <= 10) {
+    // no need to rank if there are less than 10 candidates
+    return candidates
+  }
+
+  const unscoredCandidates = candidates.filter(
+    (item) => item.score === undefined
+  )
+
+  const data = {
+    user_id: userId,
+    item_features: unscoredCandidates.reduce((acc, item) => {
+      acc[item.id] = {
+        title: item.title,
+        has_thumbnail: !!item.thumbnail,
+        has_site_icon: !!item.siteIcon,
+        saved_at: item.date,
+        site: item.siteName,
+        language: item.languageCode,
+        directionality: item.dir,
+        folder: item.folder,
+        subscription_type: item.subscription?.type,
+        author: item.author,
+        word_count: item.wordCount,
+        published_at: item.publishedAt,
+      } as Feature
+      return acc
+    }, {} as Record<string, Feature>),
+  }
+
+  const newScores = await getScores(data)
```

Comment:
Currently I will update the score for a private library item once it is saved so if this score is available, we will use it instead of calling the score API again